socialhome/streams/app/components/ReactionsBar.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+    <div class="grid-item-bar d-flex justify-content-start">
+        <slot />
+        <div class="ml-auto grid-item-reactions mt-1">
+            <b-button
+                v-if="showShares"
+                :class="{'item-reaction-shared': hasShared$, 'item-reaction': true}"
+                @click.stop.prevent="expandShares"
+            >
+                <i class="fa fa-refresh" title="Shares" aria-label="Shares"></i>
+                <span class="item-reaction-counter">{{ sharesCount$ }}</span>
+            </b-button>
+            <b-button v-if="showReplies" class="item-reaction" @click.stop.prevent="expandComments">
+                <span class="item-open-replies-action">
+                    <i class="fa fa-envelope" title="Replies" aria-label="Replies"></i>
+                    <span class="item-reaction-counter">{{ repliesCount$ }}</span>
+                </span>
+            </b-button>
+        </div>
+        <div v-if="showSharesBox" class="content-actions">
+            <b-button v-if="hasShared$" variant="secondary" @click.prevent.stop="unshare">Unshare</b-button>
+            <b-button v-if="!hasShared$" variant="secondary" @click.prevent.stop="share">Share</b-button>
+        </div>
+        <div v-if="showRepliesBox">
+            <div class="replies-container"></div>
+            <div v-if="$store.state.isUserAuthenticated" class="content-actions">
+                <b-button :href="replyUrl" variant="secondary">Reply</b-button>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+import Vue from "vue"
+import store from "streams/app/stores/applicationStore"
+
+
+export default Vue.component("reactions-bar", {
+    store,
+    props: {
+        id: {type: Number, required: true},
+        repliesCount: {type: Number, required: true},
+        sharesCount: {type: Number, required: true},
+        hasShared: {type: Boolean, required: true},
+    },
+    data() {
+        return {
+            showSharesBox: false,
+            showRepliesBox: false,
+            sharesCount$: this.sharesCount,
+            repliesCount$: this.repliesCount,
+            hasShared$: this.hasShared,
+        }
+    },
+    computed: {
+        showReplies() {
+            return this.$store.state.isUserAuthenticated || this.repliesCount$ > 0
+        },
+        showShares() {
+            return this.$store.state.isUserAuthenticated || this.sharesCount$ > 0
+        },
+        replyUrl() {
+            return Urls["content:reply"]({pk: this.id})
+        },
+    },
+    methods: {
+        expandShares() {
+            this.showSharesBox = !this.showSharesBox
+            this.showRepliesBox = this.showSharesBox ? false : this.showRepliesBox
```

Comment:
I don't think this is a good idea. Reason: replies. If there are replies, say 10 of them, clicking the share shouldn't make the replies disappear. And if it doesn't, it really doesn't make sense to make the reply button *at the end of the replies*  to disappear.

I would prefer to keep old functionality where the reaction icons just act as toggles and don't depend on each other. It would be different if they were somehow dependent on each other in concept (like in a "you can do one or the other"), but they're not, they're completely different things.

Reply:
Alright.